php/libraries/IdentifierGenerator.php

```diff
@@ -67,21 +113,6 @@ protected function createNewID(): string
             break;
         }
 
-        if (!empty($this->siteAlias) && !empty($this->projectAlias)) {
-            throw new \InvalidArgumentException(
-                'Only one of siteAlias or projectAlias should have a value'
-            );
-        }
-        if (empty($this->siteAlias) && empty($this->projectAlias)) {
-            throw new \InvalidArgumentException(
-                'Either siteAlias or projectAlias must not be empty'
-            );
-        }
-        // Assign class property 'prefix' to whichever alias is not empty.
-        !empty($this->siteAlias) ?
-            $this->prefix   = $this->siteAlias
-            : $this->prefix = $this->projectAlias;
-
```

Comment:
Removing this checkig looks like a regression. Why is it deleted?

Reply:
This check was actually causing a bug reported in the linked issued. 

This validation is now done in the child class SiteIDGenerator instead. It's more appropriate here because only PSCIDs and ExternalIDs need Site/Project aliases for their prefix. By contrast, CandIDGenerator extends IdentifierGenerator but doesn't use this functionality at all.